kf/bpf_test.go

```diff
@@ -751,32 +788,243 @@ func Test_VerifyPinnedMapVanish(t *testing.T) {
+					MapName:  "root_array",
+				},
+				Cmd:          nil,
+				RestartCount: 0,
+				hostConfig: &config.Config{
+					BpfMapDefaultPath:  "/sys/fs/bpf",
+					TcMapsRelativePath: "/tc/globals",
 				},
 			},
-			wantErr: false,
+			result: "/sys/fs/bpf/root_array",
+		},
+		{
+			name: "TcTypeMapName",
+			fields: fields{
+				Program: models.BPFProgram{
+					ProgType: models.TCType,
+					MapName:  "tc_ingress_array",
+				},
+				Cmd:          nil,
+				RestartCount: 0,
+				hostConfig: &config.Config{
+					BpfMapDefaultPath:  "/sys/fs/bpf",
+					TcMapsRelativePath: "/tc/globals",
+				},
+			},
+			result: "/sys/fs/bpf/tc/globals/tc_ingress_array",
+		},
+		{
+			name: "XdpTypeMapName_windows",
+			fields: fields{
+				Program: models.BPFProgram{
+					ProgType: models.XDPType,
+					MapName:  "root_array",
+				},
+				Cmd:          nil,
+				RestartCount: 0,
+				hostConfig: &config.Config{
+					BpfMapDefaultPath:  "c:\\sys\\fs\\bpf",
+					TcMapsRelativePath: "tc\\globals",
+				},
+			},
+			result: "c:\\sys\\fs\\bpf\\root_array",
+		},
+		{
+			name: "TcTypeMapName_windows",
+			fields: fields{
+				Program: models.BPFProgram{
+					ProgType: models.TCType,
+					MapName:  "tc_ingress_array",
+				},
+				Cmd:          nil,
+				RestartCount: 0,
+				hostConfig: &config.Config{
+					BpfMapDefaultPath:  "c:\\sys\\fs\\bpf",
+					TcMapsRelativePath: "tc\\globals",
+				},
+			},
+			result: "c:\\sys\\fs\\bpf\\tc\\globals\\tc_ingress_array",
 		},
 	}
+	counter := 0
 	for _, tt := range tests {
+		if runtime.GOOS == "windows" && counter < 2 {
+			continue
+		}
+		if runtime.GOOS != "windows" && counter >= 2 {
+			continue
+		}
 		t.Run(tt.name, func(t *testing.T) {
 			b := &BPF{
 				Program:      tt.fields.Program,
 				Cmd:          tt.fields.Cmd,
 				FilePath:     tt.fields.FilePath,
 				RestartCount: tt.fields.RestartCount,
+				hostConfig: &config.Config{
+					BpfMapDefaultPath:  "/sys/fs/bpf",
+					TcMapsRelativePath: "/tc/globals",
+				},
 			}
-			err := b.VerifyPinnedMapVanish(true)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("VerifyPinnedMapVanish() error : %w", err)
+			output := b.MapFullPath()
+			if output != tt.result {
+				t.Errorf("MapFullPath() failed")
+			}
+		})
+		counter++
+	}
+}
+func TestPrevMapFullPath(t *testing.T) {
+	type fields struct {
+		Program      models.BPFProgram
+		Cmd          *exec.Cmd
+		FilePath     string
+		RestartCount int
+		hostConfig   *config.Config
+		PrevMapName  string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		result string
+	}{
+		{
+			name: "XdpTypeMapName",
+			fields: fields{
+				Program: models.BPFProgram{
+					ProgType: models.XDPType,
+				},
+				Cmd:          nil,
+				RestartCount: 0,
+				hostConfig: &config.Config{
+					BpfMapDefaultPath:  "/sys/fs/bpf",
+					TcMapsRelativePath: "/tc/globals",
+				},
+				PrevMapName: "root_array",
+			},
+			result: "/sys/fs/bpf/root_array",
+		},
+		{
+			name: "TcTypeMapName",
```

Comment:
Inconsistent with line 971.  If you do "_windows" there then please change here to have "_linux".
```suggestion
			name: "TcTypeMapName_linux",
```

Reply:
done